File: library/src/main/java/com/panoramagl/PLConstants.java
/*
 * PanoramaGL library
 * Version 0.2 beta
 * Copyright (c) 2010 Javier Baez <javbaezga@gmail.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.panoramagl;

import com.panoramagl.enumerations.PLCubeFaceOrientation;

public class PLConstants {
    /**
     * math constants
     */

    public static final float kPI = 3.14159265358979323846f;
    public static final float kPI8 = (kPI / 8.0f);
    public static final float kPI16 = (kPI / 16.0f);
    public static final float kToDegrees = (180.0f / kPI);
    public static final float kToRadians = (kPI / 180.0f);

    /**
     * float constants
     */

    public static final float kFloatMinValue = -1000000.0f;
    public static final float kFloatMaxValue = Float.MAX_VALUE;
    public static final float kFloatUndefinedValue = -3829713706.1158636387f;

    /**
     * display constants
     */

    public static final int kMaxDisplaySize = 4096;
    public static final float kDisplayPPIBaseline = 218.104455f;

    /**
     * object constants
     */

    public static final float kDefaultAlpha = 1.0f;

    /**
     * texture constants
     */

    public static final int kTextureMaxSize = 2048;

    /**
     * renderer constants
     */

    public static final int kViewportSize = kMaxDisplaySize;
    public static final float kViewportScale = 5.12f;
    public static final float kPerspectiveAspect = 1.0f;
    public static final float kPerspectiveZNear = 0.01f;
    public static final float kPerspectiveZFar = 100.0f;

    /**
     * panorama constants
     */

    public static final float kPanoramaRadius = 1.0f;

    /**
     * cube constants
     */

    public static final int kCubeFrontFaceIndex = PLCubeFaceOrientation.PLCubeFaceOrientationFront.ordinal();
    public static final int kCubeBackFaceIndex = PLCubeFaceOrientation.PLCubeFaceOrientationBack.ordinal();
    public static final int kCubeLeftFaceIndex = PLCubeFaceOrientation.PLCubeFaceOrientationLeft.ordinal();
    public static final int kCubeRightFaceIndex = PLCubeFaceOrientation.PLCubeFaceOrientationRight.ordinal();
    public static final int kCubeUpFaceIndex = PLCubeFaceOrientation.PLCubeFaceOrientationUp.ordinal();
    public static final int kCubeDownFaceIndex = PLCubeFaceOrientation.PLCubeFaceOrientationDown.ordinal();

    /**
     * sphere constants
     */

    public static final int kDefaultSpherePreviewDivs = 50;
    public static final int kDefaultSphereDivs = 50;

    /**
     * sphere2 constants
     */

    public static final int kDefaultSphere2PreviewDivs = 30;
    public static final int kDefaultSphere2Divs = 40;

    /**
     * cylinder constants
     */

    public static final int kDefaultCylinderPreviewDivs = 60;
    public static final int kDefaultCylinderDivs = 60;
    public static final float kDefaultCylinderHeight = 3.0f;

    /**
     * animation constants
     */

    public static final float kDefaultAnimationTimerInterval = (1.0f / 45.0f);
    public static final float kDefaultAnimationTimerIntervalByFrame = (1.0f / 30.0f);
    public static final int kDefaultAnimationFrameInterval = 1;
    public static final float kCameraLookAtAnimationTimerInterval = kDefaultAnimationTimerInterval;
    public static final float kCameraFovAnimationTimerInterval = kDefaultAnimationTimerInterval;
    public static final int kCameraLookAtAnimationMaxStep = 45;
    public static final int kCameraFovAnimationMaxStep = 25;

    /**
     * rotation constants
     */

    public static final float kRotationMinValue = -180.0f;
    public static final float kRotationMaxValue = 180.0f;
    public static final float kRotationMax2Value = (kRotationMaxValue * kRotationMaxValue);

    public static final float kDefaultPitchMinRange = -90.0f;
    public static final float kDefaultPitchMaxRange = 90.0f;

    public static final float kDefaultYawMinRange = kRotationMinValue;
    public static final float kDefaultYawMaxRange = kRotationMaxValue;

    public static final float kDefaultRollMinRange = kRotationMinValue;
    public static final float kDefaultRollMaxRange = kRotationMaxValue;

    public static final float kRotationSensitivityMinValue = 1.0f;
    public static final float kRotationSensitivityMaxValue = 180.0f;
    public static final float kDefaultRotationSensitivity = 30.0f;

    /**
     * fov (field of view) constants
     */

    public static final float kFovMinValue = 0.01f;
    public static final float kFovMaxValue = 179.0f;
    public static final float kFovMax2Value = (kFovMaxValue * kFovMaxValue);

    public static final float kFovBaseline = 90.0f;
    public static final float kDefaultFov = kFovBaseline;
    public static final float kDefaultFovMinValue = 30.0f;
    public static final float kDefaultFovMaxValue = kDefaultFov;

    public static final float kFovSensitivityMinValue = 1.0f;
    public static final float kFovSensitivityMaxValue = 100.0f;
    public static final float kDefaultFovSensitivity = 30.0f;

    public static final int kDefaultMinDistanceToEnableFov = 5;

    public static final int kDefaultFovMinCounter = 3;

    /**
     * zoom constants
     */

    public static final int kDefaultZoomLevels = 2;

    /**
     * reset constants
     */

    public static final int kDefaultNumberOfTouchesForReset = 3;

    /**
     * inertia constants
     */

    public static final int kDefaultInertiaInterval = 3;

    /**
     * accelerometer constants
     */

    public static final float kDefaultAccelerometerInterval = (1.0f / 30.0f);
    public static final float kAccelerometerSensitivityMinValue = 1.0f;
    public static final float kAccelerometerSensitivityMaxValue = 10.0f;
    public static final float kDefaultAccelerometerSensitivity = kAccelerometerSensitivityMaxValue;
    public static final float kAccelerometerMultiplyFactor = 5.0f;

    /**
     * gyroscope constants
     */

    public static final float kDefaultGyroscopeInterval = (1.0f / 30.0f);
    public static final float kGyroscopeTimeConversion = (1.0f / 1000000000.0f);
    public static final float kGyroscopeMinTimeStep = (1.0f / 40.0f);

    /**
     * magnetometer constants
     */

    public static final float kDefaultMagnetometerInterval = (1.0f / 30.0f);

    /**
     * sensorial rotation constants
     */

    public static final int kSensorialRotationThreshold = 150;
    public static final int kSensorialRotationPitchErrorMargin = 5;
    public static final int kSensorialRotationYawErrorMargin = 5;

    /**
     * scrolling constants
     */

    public static final int kDefaultMinDistanceToEnableScrolling = 30;

    /**
     * drawing constants
     */

    public static final int kDefaultMinDistanceToEnableDrawing = 10;

    /**
     * shake constants
     */

    public static final int kShakeThreshold = 1300;
    public static final int kShakeDiffTime = 100;

    /**
     * transition constants
     */

    public static final float kDefaultTransitionInterval = 3.0f;
    public static final int kDefaultTransitionIterationsPerSecond = 30;

    /**
     * hotspot constants
     */

    public static final float kDefaultHotspotSize = 0.05f;
    public static final float kDefaultHotspotZPosition = kPanoramaRadius;
    public static final float kDefaultHotspotAlpha = 0.8f;
    public static final float kDefaultHotspotOverAlpha = 1.0f;
}

File: library/src/main/java/com/panoramagl/PLConstants.kt
/*
 * PanoramaGL library
 * Version 0.2 beta
 * Copyright (c) 2010 Javier Baez <javbaezga@gmail.com>
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.panoramagl

import com.panoramagl.enumerations.PLCubeFaceOrientation

object PLConstants {
    /**
     * math constants
     */
    const val kPI: Float = 3.14159265358979323846f
    const val kPI8: Float = (kPI / 8.0f)
    const val kPI16: Float = (kPI / 16.0f)
    const val kToDegrees: Float = (180.0f / kPI)
    const val kToRadians: Float = (kPI / 180.0f)

    /**
     * float constants
     */
    const val kFloatMinValue: Float = -1000000.0f
    const val kFloatMaxValue: Float = Float.MAX_VALUE
    const val kFloatUndefinedValue: Float = -3829713706.1158636387f

    /**
     * display constants
     */
    const val kMaxDisplaySize: Int = 4096
    const val kDisplayPPIBaseline: Float = 218.104455f

    /**
     * object constants
     */
    const val kDefaultAlpha: Float = 1.0f

    /**
     * texture constants
     */
    const val kTextureMaxSize: Int = 8192

    /**
     * renderer constants
     */
    const val kViewportSize: Int = kMaxDisplaySize
    const val kViewportScale: Float = 5.12f
    const val kPerspectiveAspect: Float = 1.0f
    const val kPerspectiveZNear: Float = 0.01f
    const val kPerspectiveZFar: Float = 100.0f

    /**
     * panorama constants
     */
    const val kPanoramaRadius: Float = 1.0f

    /**
     * cube constants
     */
    @JvmField
    val kCubeFrontFaceIndex: Int = PLCubeFaceOrientation.PLCubeFaceOrientationFront.ordinal
    @JvmField
    val kCubeBackFaceIndex: Int = PLCubeFaceOrientation.PLCubeFaceOrientationBack.ordinal
    @JvmField
    val kCubeLeftFaceIndex: Int = PLCubeFaceOrientation.PLCubeFaceOrientationLeft.ordinal
    @JvmField
    val kCubeRightFaceIndex: Int = PLCubeFaceOrientation.PLCubeFaceOrientationRight.ordinal
    @JvmField
    val kCubeUpFaceIndex: Int = PLCubeFaceOrientation.PLCubeFaceOrientationUp.ordinal
    @JvmField
    val kCubeDownFaceIndex: Int = PLCubeFaceOrientation.PLCubeFaceOrientationDown.ordinal

    /**
     * sphere constants
     */
    const val kDefaultSpherePreviewDivs: Int = 50
    const val kDefaultSphereDivs: Int = 50

    /**
     * sphere2 constants
     */
    const val kDefaultSphere2PreviewDivs: Int = 30
    const val kDefaultSphere2Divs: Int = 40

    /**
     * cylinder constants
     */
    const val kDefaultCylinderPreviewDivs: Int = 60
    const val kDefaultCylinderDivs: Int = 60
    const val kDefaultCylinderHeight: Float = 3.0f

    /**
     * animation constants
     */
    const val kDefaultAnimationTimerInterval: Float = (1.0f / 120.0f)
    const val kDefaultAnimationTimerIntervalByFrame: Float = (1.0f / 120.0f)
    const val kDefaultAnimationFrameInterval: Int = 1
    const val kCameraLookAtAnimationTimerInterval: Float = kDefaultAnimationTimerInterval
    const val kCameraFovAnimationTimerInterval: Float = kDefaultAnimationTimerInterval
    const val kCameraLookAtAnimationMaxStep: Int = 20
    const val kCameraFovAnimationMaxStep: Int = 10

    /**
     * rotation constants
     */
    const val kRotationMinValue: Float = -180.0f
    const val kRotationMaxValue: Float = 180.0f
    const val kRotationMax2Value: Float = (kRotationMaxValue * kRotationMaxValue)

    const val kDefaultPitchMinRange: Float = -90.0f
    const val kDefaultPitchMaxRange: Float = 90.0f

    const val kDefaultYawMinRange: Float = kRotationMinValue
    const val kDefaultYawMaxRange: Float = kRotationMaxValue

    const val kDefaultRollMinRange: Float = kRotationMinValue
    const val kDefaultRollMaxRange: Float = kRotationMaxValue

    const val kRotationSensitivityMinValue: Float = 1.0f
    const val kRotationSensitivityMaxValue: Float = 270.0f
    const val kDefaultRotationSensitivity: Float = 45.0f

    /**
     * fov (field of view) constants
     */
    const val kFovMinValue: Float = 0.01f
    const val kFovMaxValue: Float = 179.0f
    const val kFovMax2Value: Float = (kFovMaxValue * kFovMaxValue)

    const val kFovBaseline: Float = 90.0f
    const val kDefaultFov: Float = kFovBaseline
    const val kDefaultFovMinValue: Float = 30.0f
    const val kDefaultFovMaxValue: Float = kDefaultFov

    const val kFovSensitivityMinValue: Float = 1.0f
    const val kFovSensitivityMaxValue: Float = 100.0f
    const val kDefaultFovSensitivity: Float = 30.0f

    const val kDefaultMinDistanceToEnableFov: Int = 5

    const val kDefaultFovMinCounter: Int = 3

    /**
     * zoom constants
     */
    const val kDefaultZoomLevels: Int = 2

    /**
     * reset constants
     */
    const val kDefaultNumberOfTouchesForReset: Int = 3

    /**
     * inertia constants
     */
    const val kDefaultInertiaInterval: Int = 3

    /**
     * accelerometer constants
     */
    const val kAccelerometerSensitivityMinValue: Float = 1.0f
    const val kAccelerometerSensitivityMaxValue: Float = 10.0f
    const val kDefaultAccelerometerSensitivity: Float = kAccelerometerSensitivityMaxValue
    const val kAccelerometerMultiplyFactor: Float = 5.0f

    /**
     * gyroscope constants
     */
    const val kDefaultGyroscopeInterval: Float = (1.0f / 30.0f)
    const val kGyroscopeTimeConversion: Float = (1.0f / 1000000000.0f)
    const val kGyroscopeMinTimeStep: Float = (1.0f / 40.0f)

    /**
     * magnetometer constants
     */
    const val kDefaultMagnetometerInterval: Float = (1.0f / 30.0f)

    /**
     * sensorial rotation constants
     */
    const val kSensorialRotationThreshold: Int = 150
    const val kSensorialRotationPitchErrorMargin: Int = 5
    const val kSensorialRotationYawErrorMargin: Int = 5

    /**
     * scrolling constants
     */
    const val kDefaultMinDistanceToEnableScrolling: Int = 30

    /**
     * drawing constants
     */
    const val kDefaultMinDistanceToEnableDrawing: Int = 10

    /**
     * shake constants
     */
    const val kShakeThreshold: Int = 1300
    const val kShakeDiffTime: Int = 100

    /**
     * transition constants
     */
    const val kDefaultTransitionInterval: Float = 3.0f
    const val kDefaultTransitionIterationsPerSecond: Int = 30

    /**
     * hotspot constants
     */
    const val kDefaultHotspotSize: Float = 0.05f
    const val kDefaultHotspotZPosition: Float = kPanoramaRadius
    const val kDefaultHotspotAlpha: Float = 0.8f
    const val kDefaultHotspotOverAlpha: Float = 1.0f
}